tic_tac_toe/game.py

```diff
@@ -63,42 +134,80 @@ def _check_winning_combinations(board, player):
     Returns the player (winner) of any of the winning combinations is completed
     by given player, or None otherwise.
     """
-    pass
-
 
 # public interface
 def start_new_game(player1, player2):
-    """
-    Creates and returns a new game configuration.
-    """
-    pass
-
+    game = {}
+    game['player1'] = player1
+    game['player2'] = player2
+    game['board'] = [
+        ["-", "-", "-"],
+        ["-", "-", "-"],
+        ["-", "-", "-"],
+    ]
+    
+    game['next_turn'] = 'X'
+    game['winner'] = None
+    
+    return game
+ 
 
 def get_winner(game):
     """
     Returns the winner player if any, or None otherwise.
     """
-    pass
+    return game["winner"]
 
 
 def move(game, player, position):
+    if get_winner(game) != None or _board_is_full(game['board'])==True:
+        raise InvalidMovement("Game is over.")
+    if not _position_is_valid(position): 
+        raise InvalidMovement("Position out of range.")
+    if not _position_is_empty_in_board(position, game['board']):
+        raise InvalidMovement("Position already taken.")
+    if not player == game['next_turn']:
+        message= '"' + game['next_turn'] + '"'
+        raise InvalidMovement(message +" moves next.")
+        
+    game['board'][position[0]][position[1]] = player
+
+    if _check_winning_combinations(game['board'], player = "O") == "O":
+        game['winner'] = "O"
+        raise GameOver('"O" wins!')
+    elif _check_winning_combinations(game['board'], player = "X") =="X":
+        game['winner'] = "X"
+        raise GameOver('"X" wins!')
+    elif _board_is_full(game['board'])==True:
+        raise GameOver("Game is tied!")
+    else:
+        if game['next_turn']== "O":
+            game['next_turn']= "X"
+        else:
+            game['next_turn']= "O"
     """
     Performs a player movement in the game. Must ensure all the pre requisites
     checks before the actual movement is done.
     After registering the movement it must check if the game is over.
     """
-    pass
 
 
 def get_board_as_string(game):
     """
     Returns a string representation of the game board in the current state.
     """
-    pass
-
+    board = game['board']
+    lines = "--------------"
+    space = "  |  "
+    firstrow, secondrow, thirdrow = space.join(board[0]), space.join(board[1]), space.join(board[2])
+    return ("\n" +firstrow + "\n" + lines + "\n" +
+        secondrow + "\n" + lines + "\n" +
+        thirdrow + '\n')
 
 def get_next_turn(game):
     """
     Returns the player who plays next, or None if the game is already over.
     """
-    pass
+    if game['winner']!=None:
```

Comment:
I think this check should not be in the get method....this method should just return what's in game['next_turn']